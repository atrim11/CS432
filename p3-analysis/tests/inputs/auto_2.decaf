// Function definitions for testing
def int subtract(int x, int y) {
    return x - y;
}

def void printNumber(int num) {
    print_str(num);  // Error: should use print_int instead of print_str
}

def int concatStrings(int a, int b) {
    return a + b;  // Error: In Decaf, strings cannot be concatenated with '+'
}

def int main() {
    int result;
    
    result = subtract(10, 5);
    printNumber(result);  // Error: wrong type to print

    // Parameter mismatch
    result = subtract(10);  // Error: subtract expects 2 parameters, provided 1

    // Type mismatch
    result = subtract("hello", 5);  // Error: first parameter should be an int

    // Recursive issues
    result = subtract(subtract(20, 10), "str");  // Error: passing string instead of int

    // Incorrect return type
    
    // Redeclaration of variables
    a = 5;
    a = 10;  // Error: redeclaration of variable 'a' in the same scope

    // Undefined variables
    b = a + c;  // Error: 'c' is not declared
    
    // Array out of bounds
    
    arr[3] = 100;  // Error: out of bounds

    // Function with wrong return type
    
    // Control structure issues
    if (b) {  // Error: non-boolean expression
        print_str("This is wrong");
    }

    while (result + 5) {  // Error: non-boolean used in while loop
        result = result - 1;
    }

    // Incorrect use of void function

    // More mismatches and errors
    test = subtract(subtract("error", 5), 10);  // Error: multiple type mismatches
    return test;
}
