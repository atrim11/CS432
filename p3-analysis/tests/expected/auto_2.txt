Type mismatch in parameter 0 of call to 'print_str': expected str but found int on line 7
Invalid number of function arguments on line 21
Type mismatch in parameter 0 of call to 'subtract': expected int but found str on line 24
Type mismatch in parameter 1 of call to 'subtract': expected int but found str on line 27
Symbol 'a' undefined on line 32
Symbol 'a' undefined on line 33
Symbol 'b' undefined on line 36
Symbol 'a' undefined on line 36
Symbol 'c' undefined on line 36
Symbol 'arr' undefined on line 40
Symbol 'b' undefined on line 45
Type mismatch: bool expected but int found on line 49
Symbol 'test' undefined on line 56
Type mismatch in parameter 0 of call to 'subtract': expected int but found str on line 56
Symbol 'test' undefined on line 57
